src/Common/ColumnsHashingImpl.h

```diff
@@ -172,10 +194,25 @@ class HashMethodBase
         return data.hash(keyHolderGetKey(key_holder));
     }
 
+    ALWAYS_INLINE bool isNullAt(size_t row) const
+    {
+        if constexpr (nullable)
+        {
+            return assert_cast<const ColumnUInt8 &>(*null_map).getData()[row] != 0;
```

Comment:
I don't expect it to affect perf, but imo we'd better to do cast only once in ctor

Reply:
done